src/nanotron/serialize/optimizer.py

```diff
@@ -30,9 +30,9 @@ def optimizer_filename(parallel_context: ParallelContext, is_zero: bool):
         return f"{ObjectType.OPTIMIZER.value}_pp-{dist.get_rank(parallel_context.pp_pg)}-of-{parallel_context.pp_pg.size()}_tp-{dist.get_rank(parallel_context.tp_pg)}-of-{parallel_context.tp_pg.size()}_exp-{dist.get_rank(parallel_context.expert_pg)}-of-{parallel_context.expert_parallel_size}.pt"
 
 
-def lr_scheduler_filename():
+def lr_scheduler_filename(parallel_context: ParallelContext):
     """The lr_scheduler is the same for all processes."""
-    return f"{ObjectType.LR_SCHEDULER.value}.pt"
+    return f"{ObjectType.LR_SCHEDULER.value}_pp-{dist.get_rank(parallel_context.pp_pg)}-of-{parallel_context.pp_pg.size()}.pt"
```

Comment:
still need to fix this when different TPs have different number of lr_lambdas aka different number of params
So ideally we want the lr_scheduler to follow the same naming as optimizer (we can maybe make them share the same filename function)

Reply:
Now we use the same naming convention as the optimizer. Added `is_zero` flag which is obtained either from `config.optimizer.zero_stage` OR `optimizer.inherit_from(optim.ZeroDistributedOptimizer)`